spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -392,6 +362,47 @@ impl TryFrom<i32> for Offset {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+struct FixedTimeZoneOffset {
+    pub sign: i32,
+    pub hours: i32,
+    pub minutes: i32,
+}
+
+/// Parses a timezone offset from a string with either the format `+hhmm` /
+/// `-hhmm` or `+hh:mm` / `-hh:mm`. Returns a fixed timezone on success.
+fn parse_fixed_timezone_offset(input: &str) -> Result<FixedTimeZoneOffset, TzStringError> {
+    if !input.is_ascii() {
+        return Err(TzStringError::new());
+    }
+
+    // Special handling of the +/- sign is required because `-00:30` must parse
+    // to a negative offset and `i32::from_str_radix` cannot preserve the `-`
+    // sign when parsing zero.
+    let (sign, hour_str, minute_str) = match input.as_bytes() {
+        // Matches `+hhmm`
+        [b'+', _, _, _, _] => (1, &input[1..3], &input[3..5]),
+        // Matches `-hhmm`
+        [b'-', _, _, _, _] => (-1, &input[1..3], &input[3..5]),
+        // Matches `+hh:mm`
+        [b'+', _, _, b':', _, _] => (1, &input[1..3], &input[4..6]),
+        // Matches `-hh:mm`
+        [b'-', _, _, b':', _, _] => (-1, &input[1..3], &input[4..6]),
+        // Anything else is an error.
+        _ => return Err(TzStringError::new()),
+    };
```

Comment:
This is absolutely beautiful way to solve this string parsing. I'm sure it's faster than regex, but I'm also curious how the compiler optimizes this out.